docs/modules/ROOT/pages/guide-metatx.adoc

```diff
@@ -0,0 +1,349 @@
+
+```
+PRIVATE_KEY="Goerli private key"
+TEAM_API_KEY="Defender Team API key, used for uploading autotask code"
+TEAM_API_SECRET="Defender Team API secret"
+```
+
+[[create-relayer]]
+== Create Relayer
+
+First, install the necessary package:
+
+`npm install defender-relay-client`
+
+Create a script such as the following, supplying your Defender API key in your .env file and adjusting variable names as appropriate.
+
+```
+const { RelayClient } = require('defender-relay-client');
+const { appendFile, readFileSync, writeFileSync} = require('fs');
+
+async function run() {
+  require('dotenv').config();
+  const { TEAM_API_KEY: apiKey, TEAM_API_SECRET: apiSecret, AUTOTASK_ID: autotaskId } = process.env;
+  const relayClient = new RelayClient({ apiKey, apiSecret });
+
+  // create relay using defender client
+  const requestParams = {
+    name: 'MetaTxRelayer',
+    network: 'goerli',
+    minBalance: BigInt(1e17).toString(),
+  };
+  const relayer = await relayClient.create(requestParams);
+  
+  // store relayer info in file
+  writeFileSync('relay.json', JSON.stringify({
+    relayer
+  }, null, 2));
+  console.log('Relayer ID: ', relayer);
+
+  // create and save the api key to .env - needed for sending tx
+  const {relay} = JSON.parse(readFileSync('relay.json'))
+  const secretKey = await relayClient.createKey(relayer.relayerId);
+  writeFileSync('.env.relay', JSON.stringify({secretKey}, null, 2));
+}
+
+run().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
+});
+```
+
+Run with `yarn create-relay`
+
+Note that you will later need the `relayerId` from either the console log or the created `relay.json` file. You'll use this when creating the Autotask.
+
+The Relayer's API key and secret will be used whenever you would like to send a transaction via API. In the above code, it is saved to a `.env.relay` file.
+
+[[compile-using-hardhat]]
+== Compile the Contract Using Hardhat
+
+Use the following code for your Registry.sol contract:
+
+```
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/metatx/ERC2771Context.sol";
+import "@openzeppelin/contracts/metatx/MinimalForwarder.sol";
+
+contract Registry is ERC2771Context {  
+  event Registered(address indexed who, string name);
+
+  mapping(address => string) public names;
+  mapping(string => address) public owners;
+
+  constructor(MinimalForwarder forwarder) // Initialize trusted forwarder
+    ERC2771Context(address(forwarder)) {
+  }
+
+  function register(string memory name) external {
+    require(owners[name] == address(0), "Name taken");
+    address owner = _msgSender(); // Changed from msg.sender
+    owners[name] = owner;
+    names[owner] = name;
+    emit Registered(owner, name);
+  }
+}
+```
+
+Run `npx hardhat compile` to get the code ready for deployment.
+
+[[deploy-using-relayer]]
+== Deploy Using Relayer
```

Comment:
This is awesome just in itself - we should make a guide in the forum about it. We'll write a hardhat plugin for this eventually, but for now, this is great!!